src/command/Overload.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\command;
+
+use pocketmine\command\parameter\defaults\TextParameter;
+use pocketmine\command\parameter\Parameter;
+use pocketmine\command\utils\CommandException;
+use pocketmine\command\utils\InvalidCommandSyntaxException;
+use function count;
+use function implode;
+
+class Overload{
+	/** @var Command */
+	protected $command;
+
+	/** @var Parameter[] */
+	protected $parameters = [];
+
+	public function __construct(Command $command){
+		$this->command = $command;
+	}
+
+	public function addParameter(Parameter $parameter) : self{
+		if(count($this->parameters) !== 0){
+			foreach($this->parameters as $oldParameter){
+				if($parameter->getName() === $oldParameter->getName()){
+					throw new CommandException("Cannot register multiple parameters with the same name");
+				}
+			}
+		}
+		$parameter->setOverload($this);
+		$this->parameters[] = $parameter;
+		return $this;
+	}
+
+	public function getCommand() : Command{
+		return $this->command;
+	}
+
+	/**
+	 * @return Parameter[]
+	 */
+	public function getParameters() : array{
+		if(count($this->parameters) === 0){
+			return [new TextParameter("args", true)]; //Prevents parameters from being sent with empty values
```

Comment:
But some commands may not have any parameters? As far as i am aware you can send a command to the client without any parameters and it will display just fine.

Reply:
I may be understanding this wrong, actually. I'm guessing this wont be sent if there are no overloads added to the command.